### PR TITLE
System Test Review

### PR DESCRIPTION
**Pull request by EdanLevy**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -69,6 +69,7 @@ dependencies {
 
 
     implementation(libs.freemarker)
+    implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:1.26.0")
```
Comment (shalom938) [inline build.gradle.kts:73]: why is this necessary?  will it be packaged in the plugin zip? if yes then its not acceptable because it may clash with jetbrains dependencies.
in nay case , dependencies should be added to settings.gradle.kts like all our other dependencies.

File: gradle.properties [hunk 2]
```diff
@@ -41,4 +41,4 @@ systemProp.org.gradle.unsafe.kotlin.assignment = true
 kotlin.incremental.useClasspathSnapshot = false
 
 # allows unicode characters
-org.gradle.jvmargs='-Dfile.encoding=UTF-8'
\ No newline at end of file
+org.gradle.jvmargs='-Dfile.encoding=UTF-8'
```
Comment (shalom938) [inline gradle.properties:44]: please don't reformat or change line numbers when its not necessary

File: ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java [hunk 3]
```diff
@@ -70,7 +70,8 @@ public void setCurrent(@Nullable String newEnv) {
     @Override
     public void setCurrent(@NotNull String newEnv, boolean refreshInsightsView, @Nullable Runnable taskToRunAfterChange) {
 
-        Log.log(LOGGER::debug, "Setting current environment , old={},new={}", this.current, newEnv);
+        Log.test(LOGGER::info, "Setting current environment , old={}, new={}", this.current, newEnv);
+        Log.log(LOGGER::debug, "Setting current environment , old={}, new={}", this.current, newEnv);
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java:75]: why so many logs of the same message?
info will explode the user log

File: ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java [hunk 4]
```diff
@@ -115,6 +116,7 @@ public List<String> getEnvironments() {
     public void refreshNowOnBackground() {
 
         Log.log(LOGGER::trace, "Refreshing Environments on background thread.");
+        Log.test(LOGGER::info, "Firing Refreshing Environments");
         Backgroundable.ensureBackground(project, "Refreshing Environments", () -> {
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java:120]: info will explode the user log.
same for all other calls to Log.test , i'm not going to write more comments about it.

File: ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java [hunk 5]
```diff
@@ -145,8 +147,10 @@ private void refreshEnvironments() {
 
             Log.log(LOGGER::trace, "Refreshing Environments list");
             var newEnvironments = analyticsService.getEnvironments();
+//            Log.test(LOGGER::info, "refreshEnvironments: from analyticsService.getEnvironments(): {}", newEnvironments);
             if (newEnvironments != null && !newEnvironments.isEmpty()) {
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java:151]: please go over the code and remove unnecessary commented lines

File: ide-common/src/main/java/org/digma/intellij/plugin/document/DocumentInfoService.java [hunk 6]
```diff
@@ -35,7 +36,7 @@
  */
 public class DocumentInfoService {
 
-    private final Logger LOGGER = Logger.getInstance(DocumentInfoService.class);
+    private static final Logger LOGGER = Logger.getInstance(DocumentInfoService.class);
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/document/DocumentInfoService.java:40]: why should it be static?

File: ide-common/src/main/java/org/digma/intellij/plugin/editor/CurrentContextUpdater.java [hunk 7]
```diff
@@ -37,7 +37,7 @@ public class CurrentContextUpdater implements Disposable {
    keep the latest method under caret that was fired. it helps us to not call contextChange if the caret is on the same
    method as before.
     */
-    private MethodUnderCaret latestMethodUnderCaret;
+    public MethodUnderCaret latestMethodUnderCaret;
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/editor/CurrentContextUpdater.java:41]: why change to public?

File: ide-common/src/main/java/org/digma/intellij/plugin/log/Log.java [hunk 8]
```diff
@@ -23,6 +23,15 @@ public static void log(Consumer<String> consumer, String format, Object... args)
...
+    public static void test(Consumer<String> consumer, String format, Object... args) {
+        StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+        String callingMethodName = stackTraceElements[2].getMethodName();
+        String lineNumber = String.valueOf(stackTraceElements[2].getLineNumber());
+        String threadName = Thread.currentThread().getName();
+        String header = String.format("%s:%s - %s", callingMethodName, lineNumber, threadName);
+        consumer.accept(header + " - " + String.format(format.replace("{}", "%s"), args));
+    }
+
     public static void debugWithException(Logger logger,Throwable e, String format, Object... args) {
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/log/Log.java:35]: this log method is called to much from all over and will explode the user log.
getting stack trace is a heavy operation and should be called only if log level is enabled. if called with info it will explode the user log.

File: ide-common/src/main/java/org/digma/intellij/plugin/psi/LanguageServiceLocator.java [hunk 9]
```diff
@@ -16,7 +17,7 @@ public class LanguageServiceLocator {
 
     private final Project project;
 
-    private final Cache cache = new Cache();
+    public final Cache cache = new Cache();
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/psi/LanguageServiceLocator.java:21]: why public ?

File: ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt [hunk 10]
```diff
@@ -33,7 +35,7 @@ class RefreshService(private val project: Project) {
     private val isGeneralRefreshButtonEnabled = AtomicBoolean(true)
 
     companion object {
-
+        private val logger = Logger.getInstance(RefreshService::class.java)
         @JvmStatic
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt:39]: why inside companion class and not as class member?

File: ide-common/src/main/java/org/digma/intellij/plugin/service/EditorInteractionService.java [hunk 11]
```diff
@@ -21,7 +22,7 @@
  */
 public class EditorInteractionService implements CaretContextService, Disposable {
 
-    private final Logger logger = Logger.getInstance(EditorInteractionService.class);
+    private static final Logger logger = Logger.getInstance(EditorInteractionService.class);
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/service/EditorInteractionService.java:26]: why change to static?

File: ide-common/src/main/java/org/digma/intellij/plugin/service/EditorInteractionService.java [hunk 12]
```diff
@@ -41,6 +42,7 @@ public EditorInteractionService(Project project) {
     }
 
     public static CaretContextService getInstance(Project project) {
+        Log.log(logger::warn, "Getting instance of " + CaretContextService.class.getSimpleName());
         return project.getService(CaretContextService.class);
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/service/EditorInteractionService.java:46]: why log in getInstance ? it will explode the log

File: ide-common/src/main/java/org/digma/intellij/plugin/settings/SettingsState.java [hunk 13]
```diff
@@ -30,7 +30,7 @@
 public class SettingsState implements PersistentStateComponent<SettingsState>, Disposable {
 
     public static final String DEFAULT_API_URL = "https://localhost:5051";
-    public static final int DEFAULT_REFRESH_DELAY = 10;
+    public static final int DEFAULT_REFRESH_DELAY = 1; // 10; // is the real implementation
     public static final String DEFAULT_JAEGER_URL = "http://localhost:16686";
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/settings/SettingsState.java:34]: please change back to original delay

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/analytics/BackendConnectionMonitor.kt [hunk 14]
```diff
@@ -14,6 +17,7 @@ import com.intellij.util.messages.MessageBusConnection
 class BackendConnectionMonitor(val project: Project) : Disposable, AnalyticsServiceConnectionEvent {
 
     companion object {
+        private val logger = Logger.getInstance(BackendConnectionMonitor::class.java)
         @JvmStatic
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/analytics/BackendConnectionMonitor.kt:21]: why inside companion and not as class member?

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/analytics/BackendConnectionMonitor.kt [hunk 15]
```diff
@@ -32,18 +36,22 @@ class BackendConnectionMonitor(val project: Project) : Disposable, AnalyticsServ
...
         analyticsConnectionEventsConnection.dispose()
         hasConnectionError = false
     }
 
     fun isConnectionError(): Boolean {
-        return hasConnectionError
+        return if (ApplicationManager.getApplication().isUnitTestMode) false else hasConnectionError
+//        return hasConnectionError // this is the real implementation
+        
     }
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/analytics/BackendConnectionMonitor.kt:48]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/analytics/BackendConnectionMonitor.kt [hunk 16]
```diff
@@ -32,18 +36,22 @@ class BackendConnectionMonitor(val project: Project) : Disposable, AnalyticsServ
...
 
     fun isConnectionOk(): Boolean {
-        return !hasConnectionError
+        return if (ApplicationManager.getApplication().isUnitTestMode) true else !hasConnectionError
+//        return !hasConnectionError  // this is the real implementation
     }
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/analytics/BackendConnectionMonitor.kt:53]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/codelens/AbstractCodeLensService.kt [hunk 17]
```diff
@@ -147,6 +147,7 @@ abstract class AbstractCodeLensService(private val project: Project): Disposable
         // try to find a replacement for CodeVisionPassFactory.clearModificationStamp and refresh only one by one.
 
         if(project.isDisposed){
+            Log.test(logger::info, "Project already disposed")
             return
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/codelens/AbstractCodeLensService.kt:151]: this log message gives no information but that the project is disposed but not why its important to know about it

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt [hunk 18]
```diff
@@ -41,6 +42,10 @@ class CodeNavigator(val project: Project) {
         val spanIdWithoutType = CodeObjectsUtil.stripSpanPrefix(spanId)
 
         SupportedLanguages.values().forEach { language ->
+            //todo: Ask Shalom
+            if (ApplicationManager.getApplication().isUnitTestMode && language == SupportedLanguages.CSHARP) {
+                return@forEach
+            }
             val languageService = LanguageService.findLanguageServiceByName(project, language.languageServiceClassName)
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt:49]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt [hunk 19]
```diff
@@ -132,6 +146,10 @@ class CodeNavigator(val project: Project) {
         val spanIdWithoutType = CodeObjectsUtil.stripSpanPrefix(spanCodeObjectId)
 
         SupportedLanguages.values().forEach { language ->
+
+            if (ApplicationManager.getApplication().isUnitTestMode && language == SupportedLanguages.CSHARP) {
+                return@forEach
+            }
             val languageService = LanguageService.findLanguageServiceByName(project, language.languageServiceClassName)
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt:153]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt [hunk 20]
```diff
@@ -153,6 +171,10 @@ class CodeNavigator(val project: Project) {
         val methodIdWithoutType = CodeObjectsUtil.stripMethodPrefix(methodId)
 
         SupportedLanguages.values().forEach { language ->
+
+            if (ApplicationManager.getApplication().isUnitTestMode && language == SupportedLanguages.CSHARP) {
+                return@forEach
+            }
             val languageService = LanguageService.findLanguageServiceByName(project, language.languageServiceClassName)
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt:178]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt [hunk 21]
```diff
@@ -173,6 +195,9 @@ class CodeNavigator(val project: Project) {
         val spanIdWithoutType = CodeObjectsUtil.stripSpanPrefix(spanId)
 
         SupportedLanguages.values().forEach { language ->
+            if (ApplicationManager.getApplication().isUnitTestMode && language == SupportedLanguages.CSHARP) {
+                return@forEach
+            }
             val languageService = LanguageService.findLanguageServiceByName(project, language.languageServiceClassName)
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt:201]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt [hunk 22]
```diff
@@ -200,6 +225,10 @@ class CodeNavigator(val project: Project) {
             val endpointId = CodeObjectsUtil.stripEndpointPrefix(it)
 
             SupportedLanguages.values().forEach { language ->
+
+                if (ApplicationManager.getApplication().isUnitTestMode && language == SupportedLanguages.CSHARP) {
+                    return@forEach
+                }
                 val languageService = LanguageService.findLanguageServiceByName(project, language.languageServiceClassName)
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt:232]: why is it necessary? if absolutely necessary please write a comment why

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/InsightsViewService.kt [hunk 23]
```diff
@@ -47,8 +48,10 @@ class InsightsViewService(project: Project) : AbstractViewService(project) {
 
 
     companion object {
+        private val logger = Logger.getInstance(InsightsViewService::class.java)
         @JvmStatic
```
Comment (shalom938) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/InsightsViewService.kt:52]: why in companion and not as class member?

File: java/src/main/java/org/digma/intellij/plugin/idea/psi/java/JavaLanguageService.java [hunk 24]
```diff
@@ -607,7 +610,7 @@ public boolean isRelevant(VirtualFile file) {
     public boolean isRelevant(PsiFile psiFile) {
         return psiFile.isValid() &&
                 psiFile.isWritable() &&
-                projectFileIndex.isInSourceContent(psiFile.getVirtualFile()) &&
+//                projectFileIndex.isInSourceContent(psiFile.getVirtualFile()) &&
                 !projectFileIndex.isInLibrary(psiFile.getVirtualFile()) &&
```
Comment (shalom938) [inline java/src/main/java/org/digma/intellij/plugin/idea/psi/java/JavaLanguageService.java:614]: what is the change here ?

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/frameworks/QuarkusConfigureDepsService.kt [hunk 25]
```diff
@@ -33,6 +33,7 @@ class QuarkusConfigureDepsService(private val project: Project) : Disposable {
 
         @JvmStatic
         fun getInstance(project: Project): QuarkusConfigureDepsService {
+            logger.warn("Getting instance of ${QuarkusConfigureDepsService::class.simpleName}")
             return project.getService(QuarkusConfigureDepsService::class.java)
```
Comment (shalom938) [inline java/src/main/kotlin/org/digma/intellij/plugin/idea/frameworks/QuarkusConfigureDepsService.kt:37]: log message in getInstance will explode the user log

File: python/src/main/java/org/digma/intellij/plugin/psi/python/PythonSpanNavigationProvider.java [hunk 26]
```diff
@@ -59,6 +59,7 @@ public PythonSpanNavigationProvider(Project project) {
 
 
     public static PythonSpanNavigationProvider getInstance(Project project) {
+        Log.log(LOGGER::warn, "Getting instance of " + PythonSpanNavigationProvider.class.getSimpleName());
         return project.getService(PythonSpanNavigationProvider.class);
```
Comment (shalom938) [inline python/src/main/java/org/digma/intellij/plugin/psi/python/PythonSpanNavigationProvider.java:63]: log message in getInstance will explode the user log

File: src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java [hunk 27]
```diff
@@ -56,14 +57,14 @@ public InsightsMessageRouterHandler(Project project, JBCefBrowser jbCefBrowser)
         this.jbCefBrowser = jbCefBrowser;
         objectMapper = new ObjectMapper();
         objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
-        objectMapper.setDateFormat( new StdDateFormat());
+        objectMapper.setDateFormat(new StdDateFormat());
     }
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java:61]: what is the change here?

File: src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java [hunk 28]
```diff
@@ -56,14 +57,14 @@ public InsightsMessageRouterHandler(Project project, JBCefBrowser jbCefBrowser)
...
 
 
     @Override
     public boolean onQuery(CefBrowser browser, CefFrame frame, long queryId, String request, boolean persistent, CefQueryCallback callback) {
 
-        Backgroundable.executeOnPooledThread( () -> {
+        Backgroundable.executeOnPooledThread(() -> {
             try {
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java:68]: please don't reformat if not necessary

File: src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java [hunk 29]
```diff
@@ -123,7 +124,7 @@ public boolean onQuery(CefBrowser browser, CefFrame frame, long queryId, String
                     default -> throw new IllegalStateException("Unexpected value: " + action);
                 }
 
-                stopWatchStop(stopWatch, time -> Log.log(LOGGER::trace, "action {} took {}",action, time));
+                stopWatchStop(stopWatch, time -> Log.log(LOGGER::trace, "action {} took {}", action, time));
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java:128]: please don't reformat if not necessary

File: src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java [hunk 30]
```diff
@@ -254,17 +255,21 @@ void pushInsights(List<CodeObjectInsight> insights, List<Span> spans, String ass
...
+        Log.test(LOGGER::info, "sending INSIGHTS/SET_DATA message");
         try {
-            jbCefBrowser.getCefBrowser().executeJavaScript(
+//            JCefBrowserUtil.postJSMessage(objectMapper.writeValueAsString(message), jbCefBrowser);
+            CefBrowser brow = jbCefBrowser.getCefBrowser();
+                    brow.executeJavaScript(
                     "window.postMessage(" + objectMapper.writeValueAsString(message) + ");",
-                    jbCefBrowser.getCefBrowser().getURL(),
+                    brow.getURL(),
                     0);
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java:273]: why is this change necessary?

File: src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java [hunk 31]
```diff
@@ -276,9 +281,11 @@ void emptyInsights() {
...
         var message = new SetInsightsDataMessage("digma", "INSIGHTS/SET_DATA", InsightsPayload.EMPTY_INSIGHTS);
         try {
-            jbCefBrowser.getCefBrowser().executeJavaScript(
+            CefBrowser brow = jbCefBrowser.getCefBrowser();
+            String url = brow.getURL();
+            brow.executeJavaScript(
                     "window.postMessage(" + objectMapper.writeValueAsString(message) + ");",
-                    jbCefBrowser.getCefBrowser().getURL(),
+                    url,
                     0);
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/insights/InsightsMessageRouterHandler.java:289]: why is this change necessary?

File: system-test/src/test/resources/TestFile.kt [hunk 32]
```diff
@@ -0,0 +1,18 @@
...
+    
+    fun relatedSpan1() { }
+    
+    fun relatedSpan2() { }
+    
+    fun relatedEndpoint1() { }
+    
+    fun relatedEndpoint2() { }
+
+}
```
Comment (shalom938) [inline system-test/src/test/resources/TestFile.kt:18]: does the test also tests spans and endpoints?

File: system-test/src/test/resources/TestFile2.java [hunk 33]
```diff
@@ -0,0 +1,27 @@
...
+    public void relatedSpan2() {
+    }
+
+    public void relatedEndpoint1() {
+    }
+
+    public void relatedEndpoint2() {
+    }
+
+}
```
Comment (shalom938) [inline system-test/src/test/resources/TestFile2.java:27]: does the test also tests spans and endpoints?

**Review by shalom938 — COMMENTED** [review]
too many log messages.
too many commented lines and unnecessary reformats.
can the test also work with kotlin files?
the test should run for all the intellij versions we support not only the default one.
how to integrate the test into the build system? github actions?

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/Log.java [hunk 34]
```diff
@@ -0,0 +1,16 @@
...
+    public static void test(Consumer<String> consumer, String format, Object... args) {
+        StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+        String callingMethodName = stackTraceElements[2].getMethodName();
+        String lineNumber = String.valueOf(stackTraceElements[2].getLineNumber());
+        String threadName = Thread.currentThread().getName();
+        String header = String.format("%s:%s - %s", callingMethodName, lineNumber, threadName);
+        consumer.accept(header + " - " + String.format(format.replace("{}", "%s"), args));
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (shalom938) [reply analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/Log.java:16]: please remove empty class

File: ide-common/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsService.java [hunk 35]
```diff
@@ -118,6 +120,7 @@ public class AnalyticsService implements Disposable {
     private AnalyticsProvider analyticsProviderProxy;
 
     public AnalyticsService(@NotNull Project project) {
+        Log.test(LOGGER:: info, "Initializing AnalyticsService");
         //initialize BackendConnectionMonitor when starting, so it is aware early on connection statuses
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsService.java:124]: why test log?

File: ide-common/src/main/java/org/digma/intellij/plugin/common/Backgroundable.java [hunk 36]
```diff
@@ -23,9 +23,11 @@ private Backgroundable() {
     public static void ensureBackground(Project project, String name, Runnable task) {
 
         Log.log(LOGGER::trace, "Request to call task '{}'", name);
+        Log.test(LOGGER::info, "Requesting to fire task '{}'", name);
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/common/Backgroundable.java:27]: remove test log

File: ide-common/src/main/java/org/digma/intellij/plugin/common/Backgroundable.java [hunk 37]
```diff
@@ -23,9 +23,11 @@ private Backgroundable() {
...
 
         if (EDT.isEdt()) {
             Log.log(LOGGER::trace, "Executing task '{}' in background thread", name);
+            Log.test(LOGGER::info, "Firing task '{}' in background thread", name);
             new Task.Backgroundable(project, name) {
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/common/Backgroundable.java:31]: remove test log

File: ide-common/src/main/java/org/digma/intellij/plugin/common/Backgroundable.java [hunk 38]
```diff
@@ -34,6 +36,7 @@ public void run(@NotNull ProgressIndicator indicator) {
             }.queue();
         } else {
             Log.log(LOGGER::trace, "Executing task '{}' in current thread", name);
+            Log.test(LOGGER::info, "Executing task '{}' in current thread", name);
             runWithErrorReporting(project, name, task);
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/common/Backgroundable.java:40]: remove test log

File: ide-common/src/main/java/org/digma/intellij/plugin/document/DocumentInfoService.java [hunk 39]
```diff
@@ -7,6 +7,7 @@
 import com.intellij.psi.PsiFile;
 import org.apache.commons.lang3.mutable.MutableInt;
 import org.digma.intellij.plugin.analytics.AnalyticsService;
+import org.digma.intellij.plugin.common.DumbAwareNotifier;
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/document/DocumentInfoService.java:10]: unused import

File: ide-common/src/main/java/org/digma/intellij/plugin/htmleditor/DigmaHTMLEditorProvider.java [hunk 40]
```diff
@@ -36,6 +39,7 @@ public DigmaHTMLEditorProvider(Project project) {
     }
 
     public static DigmaHTMLEditorProvider getInstance(Project project){
+        Log.log(LOGGER::warn, "Getting instance of " + DigmaHTMLEditorProvider.class.getSimpleName());
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/htmleditor/DigmaHTMLEditorProvider.java:42]: why log in getInstance ?

File: ide-common/src/main/java/org/digma/intellij/plugin/psi/LanguageServiceLocator.java [hunk 41]
```diff
@@ -3,6 +3,7 @@
 import com.intellij.lang.Language;
 import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.project.Project;
+import org.digma.intellij.plugin.htmleditor.DigmaHTMLEditorProvider;
```
Comment (shalom938) [inline ide-common/src/main/java/org/digma/intellij/plugin/psi/LanguageServiceLocator.java:6]: unused import

File: ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt (hunk 10 above)
Comment (EdanLevy) [reply ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt:38]: we wanted to log the get instance. so we had to put it in the companion object

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt (hunk 18 above)
Comment (EdanLevy) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/navigation/codenavigation/CodeNavigator.kt:49]: in unit test mode, the call to LanguageService.findLanguageServiceByName with language CSHARP, crashes the plugin.

File: system-test/src/test/resources/TestFile.kt (hunk 32 above)
Comment (EdanLevy) [reply system-test/src/test/resources/TestFile.kt:18]: No, those are for the insights mock data. those methods id's are placed in the Insight as relatedSpan and Endpoint but they real location in the test files for us to navigate to them

File: system-test/src/test/resources/TestFile2.java (hunk 33 above)
Comment (EdanLevy) [reply system-test/src/test/resources/TestFile2.java:27]: No, those are for the insights mock data. those methods id's are placed in the Insight as relatedSpan and Endpoint but they real location in the test files for us to navigate to them

**Review by shalom938 — APPROVED** [review]
